lib/std/fs/test.zig

```diff
@@ -1068,3 +1068,72 @@ test "chown" {
     defer dir.close();
     try dir.chown(null, null);
 }
+
+test "File.metadata" {
+    var tmp = tmpDir(.{});
+    defer tmp.cleanup();
+
+    const file = try tmp.dir.createFile("test_file", .{ .read = true });
+    defer file.close();
+
+    const metadata = try file.metadata();
+    try testing.expect(metadata.kind() == .File);
+    try testing.expect(metadata.size() == 0);
+    _ = metadata.time(.accessed);
+}
+
+test "Metadata.permissions" {
+    if (builtin.os.tag == .wasi)
+        return error.SkipZigTest;
+
+    var tmp = tmpDir(.{});
+    defer tmp.cleanup();
+
+    const file = try tmp.dir.createFile("test_file", .{ .read = true });
+    defer file.close();
+
+    const metadata = try file.metadata();
+    var permissions = metadata.permissions();
+
+    try testing.expect(!permissions.readonly());
+    permissions.setReadonly(true);
+    try testing.expect(permissions.readonly());
+
+    try file.setPermissions(permissions);
+
+    const new_metadata = try file.metadata();
+    const new_permissions = new_metadata.permissions();
+
+    try testing.expect(new_permissions.readonly());
```

Comment:
```suggestion
    try testing.expect(new_permissions.readonly());

    // Must set it back to non-read-only in order to delete it
    permissions.setReadonly(false);
    try file.setPermissions(permissions);
```

This will fix the test on Windows (it was running into an unexpected `CANNOT_DELETE` status during the `defer tmp.cleanup()`). I'll make a separate PR to fix the unexpected status part of this.

Reply:
Thanks for spotting this :)